infra/Pos/Reporting/Methods.hs

```diff
@@ -73,9 +203,33 @@ type MonadReporting ctx m =
        , HasCompileInfo
        )
 
-----------------------------------------------------------------------------
--- Node-specific
-----------------------------------------------------------------------------
+
+-- Common code across node sending: tries to send logs to at least one
+-- reporting server.
+sendReportNodeImpl
+    :: forall ctx m. (MonadReporting ctx m)
+    => Maybe Text -> ReportType -> m ()
+sendReportNodeImpl rawLogs reportType = do
+    servers <- view (reportingContext . reportServers)
+    if null servers
+    then onNoServers
+    else do
+        let withPath :: ([FilePath] -> m ()) -> m ()
+            withPath =
+                maybe (\a -> a [])
+                      (\t cont -> withTempLogFile t $ \file -> cont [file])
+                      rawLogs
+        -- logFile is either [a] or [].
```

Comment:
FYI: it's called `Maybe a`.

Reply:
Your irony is inappropriate, because code here is continuation based and `sendReport` interface requires list, so i couldn't use any `Maybe` related methods here.

Reply:
Why not write it like this?
```
+        let withPath :: (Maybe FilePath -> m ()) -> m ()
+            withPath =
+                maybe (\a -> a Nothing)
+                      (\t cont -> withTempLogFile t $ \file -> cont (Just file))
+                      rawLogs

+        withPath $ \(logFile :: Maybe FilePath) -> do
+            errors <-
+                fmap lefts $ forM servers $
+                try . sendReport (maybeToList logFile) reportType "cardano-node" . toString
```

Reply:
Extra intermediate conversion to `Maybe` 🤷‍♂️